kratos/elements/distance_calculation_flux_based_element.h

```diff
@@ -0,0 +1,359 @@
+
+    static constexpr unsigned int NumNodes = TNumNodes;
+
+    static constexpr unsigned int BlockSize = 1;
+
+    static constexpr unsigned int LocalSize = NumNodes * BlockSize;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constuctor.
+    /**
+    * @param NewId Index number of the new element (optional)
+    */
+    DistanceCalculationFluxBasedElement(IndexType NewId = 0) :
+        Element(NewId)
+    {}
+
+    /// Constructor using an array of nodes.
+    /**
+    * @param NewId Index of the new element
+    * @param ThisNodes An array containing the nodes of the new element
+    */
+    DistanceCalculationFluxBasedElement(IndexType NewId, const NodesArrayType& ThisNodes):
+        Element(NewId, ThisNodes)
+    {}
+
+    /// Constructor using a geometry object.
+    /**
+    * @param NewId Index of the new element
+    * @param pGeometry Pointer to a geometry object
+    */
+    DistanceCalculationFluxBasedElement(IndexType NewId, GeometryType::Pointer pGeometry):
+        Element(NewId, pGeometry)
+    {}
+
+    /// Constuctor using geometry and properties.
+    /**
+    * @param NewId Index of the new element
+    * @param pGeometry Pointer to a geometry object
+    * @param pProperties Pointer to the element's properties
+    */
+    DistanceCalculationFluxBasedElement(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties) :
+        Element(NewId, pGeometry, pProperties)
+    {}
+
+    /// Destructor.
+    virtual ~DistanceCalculationFluxBasedElement() override
+    {}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Create a new element of this type
+    /**
+    * Returns a pointer to a new DistanceCalculationFluxBasedElement element, created using given input.
+    * @param NewId the ID of the new element
+    * @param ThisNodes the nodes of the new element
+    * @param pProperties the properties assigned to the new element
+    * @return a Pointer to the new element
+    */
+    Element::Pointer Create(IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        Properties::Pointer pProperties)const override
+    {
+        return Element::Pointer(new DistanceCalculationFluxBasedElement(NewId, GetGeometry().Create(ThisNodes), pProperties));
+    }
+
+    /// Create a new element of this type using given geometry
+    /**
+    * Returns a pointer to a new DistanceCalculationFluxBasedElement element, created using given input.
+    * @param NewId the ID of the new element
+    * @param pGeom a pointer to the geomerty to be used to create the element
+    * @param pProperties the properties assigned to the new element
+    * @return a Pointer to the new element
+    */
+    Element::Pointer Create(IndexType NewId,
+        GeometryType::Pointer pGeom,
+        Properties::Pointer pProperties) const override 
+    {
+        return Element::Pointer(new DistanceCalculationFluxBasedElement(NewId, pGeom, pProperties));
```

Comment:
`make_intrusive` function defined in `smart_pointers.h` should be used instead.

Reply:
ok